### ISSUE TITLE
Add support for Fine-Grained (aws_elasticsearch_domain)

### ISSUE THREAD

**Issue by jujugrrr** [terraform-providers/terraform-provider-aws#12054]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

AWS released a new feature for AWS ES Domain. You can now have fine-grained security, which is a great feature, specially for Kibana access.
https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/fgac.html

### New or Affected Resource(s)

* aws_elasticsearch_domain

### Potential Terraform Configuration

Could be the same as the cognito_options:

```hcl
resource "aws_elasticsearch_domain" "example" {
  # .. other configuration ...
  fine_grained_options {
    enabled = true
   # 
    master_user = "admin"
    master_password = "secret"
  }
}

resource "aws_elasticsearch_domain" "example" {
  # .. other configuration ...
  fine_grained_options {
    enabled = true
    iam_arn = "arn:xxxxx::role/aws-es-admin-role"
  }
}
```

### References

* #0000

**Comment by tomelliff** [comment]
I think this is configured at the Elasticsearch level (eg via the UI or the Elasticsearch API) rather than at the AWS Elasticsearch service level so I don't think this provider can help here.

https://opendistro.github.io/for-elasticsearch-docs/docs/security-access-control/ shows how to configure it but that's just general Open Distro for Elasticsearch stuff.

**Comment by jujugrrr** [comment]
Hi Tom, it's definetly an option you can tick through the AWS console, so
are the master user details.  I haven't look at AWS API support for it
though

**Comment by bflad** [comment]
The [AWS Elasticsearch API Reference contains a new AdvancedSecurityOptions structure](https://docs.aws.amazon.com/elasticsearch-service/latest/developerguide/es-configuration-api.html#es-configuration-api-datatypes-advancedsec), which looks like it would be the relevant piece for the Terraform resource. My suggestion for anyone potentially implementing this would be to follow the API structure, e.g. in Terraform configuration

```hcl
resource "aws_elasticsearch_domain" "example" {
  # ... other configuration ...

  advanced_security_options {
    enabled = true
    internal_user_database_enabled = true

    master_user_options {
      master_user_arn = ""
      master_user_name = ""
      master_user_password = ""
    }
  }
}
```

**Comment by anGie44** [comment]
This feature has been merged and will release with `v2.69.0` of the Terraform AWS Provider, expected later this week.
